### PR TITLE
Add shellThemeFromScheme function to lib-contrib

### PR DESCRIPTION
**Pull request by xlambein**
Adds a Nix function which generates a shell script that applies a given color scheme to the current shell.

Inspired by https://git.sr.ht/~misterio/shellcolord and https://github.com/Misterio77/nix-colors/pull/14

### CODE REVIEW

**Comment by Misterio77** [comment]
Thank you!

This exactly what I had in mind. Works flawlessly on all shells I've tested.

I'm just pushing a quick fix for also apply the color to tmux' borders (it's been lying around but I forgot to update shellcolord with it). GNU screen is working nicely as well.

File: lib/contrib/shell-theme.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ pkgs }: { scheme }:
+# Source: https://git.sr.ht/~misterio/shellcolord
+with scheme.colors;
+pkgs.writeShellScript "shell-theme-${scheme.slug}.sh" ''
+  if [ "''${TERM%%[-.]*}" = "screen" ]; then
+      apply_color() { echo -ne "\033P\033]$@\007\033\\"; }
+  else
```
Comment (Misterio77) [inline lib/contrib/shell-theme.nix:7]:
```suggestion
  elif [ -n "$TMUX" ]; then
      apply_color() { echo -ne "\033Ptmux;\033\033]$@\033\033\\\033\\"; }
  else
```

**Comment by xlambein** [comment]
Cool! :) I applied your changes, do you want me to squash the commits or something?

**Comment by Misterio77** [comment]
Sorry for the delay, totally missed the blue dot

I usually squash, but I can do that with just one click, no worries!

Thanks a lot again <3

**Review by Misterio77 — APPROVED** [review]

**Pull request merged**